### DISCUSSION TITLE
c.File absolute path 404 Not found

### DISCUSSION THREAD

**Discussion by navaneeth-dev** [pocketbase/pocketbase#2399] [Q&A] [answered]
I want to send a file from the parent dir of pocketbase. I am authenticated also so that is not the issue

Error response:

```json
{
	"code": 404,
	"message": "Not Found.",
	"data": {}
}
```

Minimal reproduceable code:
```go
package main

import (
	"log"
	"net/http"

	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/apis"
	"github.com/pocketbase/pocketbase/core"
)
func main() {
	app := pocketbase.New()

	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
		e.Router.AddRoute(echo.Route{
			Method: http.MethodPost,
			Path:   "/api/create-build",
			Handler: func(c echo.Context) error {
				return c.File("/etc/hosts")
			},
			Middlewares: []echo.MiddlewareFunc{
				apis.ActivityLogger(app),
				apis.RequireRecordAuth(),
			},
		})

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}
```

EDIT: in echo v4 it works fine but not in pocketbase

**Comment by ganigeorgiev** [comment, marked as answer]
This is expected. The default fs in echo v5 is the current working directory, aka. `os.DirFS(".")` (https://github.com/labstack/echo/blob/v5_alpha/echo.go#L87-L93).

I'm not sure what is the behavior in v4 but it is possible that it has changed since I think the go dir fs abstraction was introduced later (maybe in go 1.16+?).

If you want to serve files from the root system directory, you can try specifying an explicitly set `os.DirFS` via `c.FileFS`, something like:

```go
c.FileFS("etc/hosts", os.DirFS("/"))
```
_Note that `os.DirFS("/")` may not work on some platforms (you can also check https://github.com/golang/go/issues/44279)._
